### PR TITLE
Remove unimplemented syntax

### PR DESCRIPTION
**Pull request by ykmr1224**
### Description
- Remove unimplemented syntax so that frontend can utilize the .g4 files for validation/auto-complete.
- I considered comment-out, but ended up deleting them as we can refer this PR or [file in SQL plugin](https://github.com/opensearch-project/sql/blob/main/ppl/src/main/antlr/OpenSearchPPLParser.g4) to see the original implementation, and it became a bit messy when I commented out them with TODO tag.

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by penghuo — APPROVED** [review]
LGTM!

**Review by noCharger — APPROVED** [review]
Thanks for the change! Curious how to identify unimplemented ones? maybe add some test?

**Review by YANG-DB — APPROVED** [review]

**Comment by ykmr1224** [comment]
> Thanks for the change! Curious how to identify unimplemented ones? maybe add some test?

Some commands had implementation which throws error, so I referred them first. I found some more unimplemented ones, and I verified those syntax fails by running it in test.

File: ppl-spark-integration/src/main/antlr4/OpenSearchPPLParser.g4 [hunk 1]
```diff
@@ -281,24 +263,17 @@ logicalExpression
...
    | primaryExpression                                                                          # valueExpressionDefault
-   | positionFunction                                                                           # positionFunctionCall
-   | extractFunction                                                                            # extractFunctionCall
-   | getFormatFunction                                                                          # getFormatFunctionCall
-   | timestampFunction                                                                          # timestampFunctionCall
    | LT_PRTHS valueExpression RT_PRTHS                                                          # parentheticValueExpr
    ;
 
 primaryExpression
-   : evalFunctionCall
```
Comment (LantaoJin) [inline ppl-spark-integration/src/main/antlr4/OpenSearchPPLParser.g4:299]: This should be a mistaken removing which cause ppl `source=t a = 1` failed.

**Pull request merged**